core/services/offchainreporting2/delegate.go

```diff
@@ -131,75 +132,132 @@ func (d Delegate) ServicesForSpec(jobSpec job.Job) (services []job.Service, err
-			LocalConfig:                  lc,
-			Logger:                       ocrLogger,
-			MonitoringEndpoint:           d.monitoringEndpointGen.GenMonitoringEndpoint(spec.ContractID),
-			OffchainConfigDigester:       offchainConfigDigester,
-			OffchainKeyring:              kb,
-			OnchainKeyring:               kb,
-			ReportingPluginFactory:       numericalMedianFactory,
-		})
-		if err != nil {
-			return nil, errors.Wrap(err, "error calling NewOracle")
-		}
-		services = append(services, oracle)
+	runResults := make(chan pipeline.Run, d.cfg.JobPipelineResultWriteQueueDepth())
+	juelsPerFeeCoinPipelineSpec := pipeline.Spec{
+		ID:           jobSpec.ID,
+		DotDagSource: spec.JuelsPerFeeCoinPipeline,
+		CreatedAt:    time.Now(),
+	}
+	numericalMedianFactory := median.NumericalMedianFactory{
+		ContractTransmitter: ocr2Provider.MedianContract(),
+		DataSource: ocrcommon.NewDataSourceV2(d.pipelineRunner,
+			jobSpec,
+			*jobSpec.PipelineSpec,
+			loggerWith,
+			runResults,
+		),
+		JuelsPerFeeCoinDataSource: ocrcommon.NewInMemoryDataSource(d.pipelineRunner, jobSpec, juelsPerFeeCoinPipelineSpec, loggerWith),
+		ReportCodec:               ocr2Provider.ReportCodec(),
+		Logger:                    ocrLogger,
+	}
+
+	jobSpec.PipelineSpec.JobName = jobSpec.Name.ValueOrZero()
+	jobSpec.PipelineSpec.JobID = jobSpec.ID
+	oracle, err := libocr2.NewOracle(libocr2.OracleArgs{
+		BinaryNetworkEndpointFactory: peerWrapper.Peer2,
+		V2Bootstrappers:              bootstrapPeers,
+		ContractTransmitter:          ocr2Provider.ContractTransmitter(),
+		ContractConfigTracker:        tracker,
+		Database:                     ocrDB,
+		LocalConfig:                  lc,
+		Logger:                       ocrLogger,
+		MonitoringEndpoint:           d.monitoringEndpointGen.GenMonitoringEndpoint(spec.ContractID),
+		OffchainConfigDigester:       offchainConfigDigester,
+		OffchainKeyring:              kb,
+		OnchainKeyring:               kb,
+		ReportingPluginFactory:       numericalMedianFactory,
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "error calling NewOracle")
+	}
 
-		// RunResultSaver needs to be started first so its available
-		// to read odb writes. It is stopped last after the Oracle is shut down
-		// so no further runs are enqueued and we can drain the queue.
-		services = append([]job.Service{ocrcommon.NewResultRunSaver(
+	// RunResultSaver needs to be started first so its available
+	// to read odb writes. It is stopped last after the Oracle is shut down
+	// so no further runs are enqueued and we can drain the queue.
+	return []job.Service{
+		ocrcommon.NewResultRunSaver(
 			runResults,
 			d.pipelineRunner,
 			make(chan struct{}),
-			loggerWith,
-		)}, services...)
+			loggerWith),
+		&oracleService{
+			provider:      ocr2Provider,
+			contractReady: contractReady,
+			oracle:        oracle,
+		},
+	}, nil
+}
+
+type oracleService struct {
+	provider      types.OCR2Provider
+	contractReady chan struct{}
```

Comment:
Could be read only on this end.
```suggestion
	contractReady <-chan struct{}
```